### PR TITLE
[pytorch][build][sagemaker] PyTorch 1.10.2 SM training re-release, fixing smdebug perf degradation

### PR DESCRIPTION
**Pull request by ShiboXing**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
PT-1.10 SM training re-release for fixing the pytorch performance degradation issue

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"` commit: [5120423](https://github.com/aws/deep-learning-containers/pull/2219/commits/51204233a538a263c179befa7853d1394e7f8e31)
- [x] Revision B: `sagemaker_remote_tests = "rc"` commit: [87017fd](https://github.com/aws/deep-learning-containers/pull/2219/commits/87017fdffd0b48c072cad0f8269256bf9a98844e)
- [x] Revision C: `sagemaker_remote_tests = "efa"` commit: [7081297](https://github.com/aws/deep-learning-containers/pull/2219/commits/7081297cb626db6b04690446cbaa4ad556a06f7a)

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ydaiming** [comment]
Could you please disable these tests for 1.10 and above?
https://github.com/aws/deep-learning-containers/blob/master/test/dlc_tests/release_candidate_integration/test_sm_profiler.py#L35

File: pytorch/training/docker/1.10/py3/cu113/Dockerfile.sagemaker.gpu [hunk 1]
```diff
@@ -73,6 +73,8 @@ RUN pip install --no-cache-dir -U \
     "shap>=0.39,<1" \
  && pip uninstall -y torch \
  && pip install --no-cache-dir -U ${PT_TRAINING_URL}
```
Comment (ydaiming) [inline pytorch/training/docker/1.10/py3/cu113/Dockerfile.sagemaker.gpu:75]: You need a backslash `\` at the end of the line.

File: pytorch/buildspec-1-10.yml [hunk 2]
```diff
@@ -46,22 +46,94 @@ context:
       target: deep_learning_container.py
 
 images:
-  # BuildEC2CPUPTTrainPy3DockerImage:
+  BuildEC2CPUPTTrainPy3DockerImage:
+    <<: *TRAINING_REPOSITORY
+    build: &PYTORCH_CPU_TRAINING_PY3 false
+    enable_test_promotion: true
```
Comment (ohadkatz) [inline pytorch/buildspec-1-10.yml:52]: You can set this value to false if SM Training is the only requirement for testing.

**Review by ohadkatz — APPROVED** [review]

**Review by josephevans — APPROVED** [review]

**Pull request merged**